lib/Html.ml

```diff
@@ -7,19 +7,42 @@ let is_self_closing_tag = function
       true
   | _ -> false
 
-(* Based on https://github.com/facebook/react/blob/97d75c9c8bcddb0daed1ed062101c7f5e9b825f4/packages/react-dom-bindings/src/server/escapeTextForBrowser.js#L51-L98 *)
-(* https://discuss.ocaml.org/t/html-encoding-of-string/4289/4 *)
-let encode s =
-  let buffer = Buffer.create (String.length s * 2) in
-  s
-  |> String.iter (function
-       | '&' -> Buffer.add_string buffer "&amp;"
-       | '<' -> Buffer.add_string buffer "&lt;"
-       | '>' -> Buffer.add_string buffer "&gt;"
-       | '"' -> Buffer.add_string buffer "&quot;"
-       | '\'' -> Buffer.add_string buffer "&#x27;"
-       | c -> Buffer.add_char buffer c);
-  Buffer.contents buffer
+(* https://github.com/andreypopp/reactor/blob/main/htmlgen/htmlgen.ml#L34-L68 *)
+let escape_and_add b s =
+  let adds = Buffer.add_string in
+  let len = String.length s in
+  let max_idx = len - 1 in
+  let flush b start i =
+    if start < len then Buffer.add_substring b s start (i - start)
+  in
+  let rec loop start i =
+    if i > max_idx then flush b start i
+    else
+      let next = i + 1 in
+      match String.get s i with
+      | '&' ->
+          flush b start i;
+          adds b "&amp;";
+          loop next next
+      | '<' ->
+          flush b start i;
+          adds b "&lt;";
+          loop next next
+      | '>' ->
+          flush b start i;
+          adds b "&gt;";
+          loop next next
+      | '\'' ->
+          flush b start i;
+          adds b "&apos;";
+          loop next next
```

Comment:
The case of `&apos` wasn't there before, which honestly looked like a bug, but this can be a breaking change indeed.